src/System.Security.Cryptography.Xml/tests/EncryptionPropertyCollectionTest.cs

```diff
@@ -0,0 +1,150 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Xml;
+using Xunit;
+
+namespace System.Security.Cryptography.Xml.Tests
+{
+    public class EncryptionPropertyCollectionTest
+    {
+        [Fact]
+        public void Ctor_Default()
+        {
+            EncryptionPropertyCollection encPropertyCollection = new EncryptionPropertyCollection();
+            Assert.Equal(0, encPropertyCollection.Count);
+            Assert.False(encPropertyCollection.IsFixedSize);
+            Assert.False(encPropertyCollection.IsReadOnly);
+            Assert.False(encPropertyCollection.IsSynchronized);
+            Assert.NotNull(encPropertyCollection.SyncRoot);
+        }
+
+        [Fact]
+        public void Add_OneEncryptionProperty()
+        {
+            EncryptionPropertyCollection encPropertyCollection = new EncryptionPropertyCollection();
+            EncryptionProperty encProperty = new EncryptionProperty();
+            encPropertyCollection.Add(encProperty);
+            Assert.Equal(1, encPropertyCollection.Count);
+            Assert.NotNull(encPropertyCollection.Item(0));
+            Assert.Equal(encProperty, encPropertyCollection.Item(0));
+        }
+
+        [Fact]
+        public void Contains_True()
+        {
+            EncryptionPropertyCollection encPropertyCollection = new EncryptionPropertyCollection();
+            EncryptionProperty encProperty = new EncryptionProperty();
+            encPropertyCollection.Add(encProperty);
+            Assert.True(encPropertyCollection.Contains(encProperty));
+        }
+
+        [Fact]
+        public void Contains_False()
+        {
+            EncryptionPropertyCollection encPropertyCollection = new EncryptionPropertyCollection();
+            EncryptionProperty encProperty = new EncryptionProperty();
+            Assert.False(encPropertyCollection.Contains(encProperty));
+        }
+
+        [Fact]
+        public void IndexOf_Contains()
+        {
+            EncryptionPropertyCollection encPropertyCollection = new EncryptionPropertyCollection();
+            EncryptionProperty encProperty = new EncryptionProperty();
+            encPropertyCollection.Add(encProperty);
+            Assert.Equal(0, encPropertyCollection.IndexOf(encProperty));
+        }
+
+        [Fact]
+        public void IndexOf_DoesNotContain()
+        {
+            EncryptionPropertyCollection encPropertyCollection = new EncryptionPropertyCollection();
+            EncryptionProperty encProperty = new EncryptionProperty();
+            Assert.Equal(-1, encPropertyCollection.IndexOf(encProperty));
+        }
+
+        [Fact]
+        public void Insert()
+        {
+            EncryptionPropertyCollection encPropertyCollection = new EncryptionPropertyCollection();
+            EncryptionProperty encProperty = new EncryptionProperty();
+            encPropertyCollection.Insert(0, encProperty);
```

Comment:
With Insert I think it might be useful to check inserting in minimum 3 cases:
- [x] empty collection
- [ ] non-empty collection, inserting as first element
- [ ] non-empty collection, inserting somewhere in the middle
- [ ] non-empty collection, inserting as last element